### ISSUE TITLE
Can you add support for more languages?

### ISSUE THREAD

**Issue by gnu-user** [anvaka/common-words#4]
Love your project, very interesting read! Would it be possible for you to add support for additional languages such as Lisp, SQL, Clojure, Rust, etc?

Thanks!

**Comment by mbStavola** [comment]
Kotlin and Rust would be cool!

**Comment by dooglio** [comment]
How about CMake?

**Comment by chrisetheridge** [comment]
Clojure / ClojureScript would be awesome!

**Comment by anvaka** [comment]
Absolutely! 

I added sql (all files with `.sql` extension): https://anvaka.github.io/common-words/#?lang=sql
And rust (all files with `.rs` extension): https://anvaka.github.io/common-words/#?lang=rust

What would be the most accurate file extension for Clojure/Lisp/Kotlin/CMake?

**Comment by quangdatv** [comment]
How about Swift and Obj-C?

**Comment by chrisetheridge** [comment]
Clojure / ClojureScript files all use the extension of `.clj` and `.cljs` respectively.

**Comment by rmflight** [comment]
How about .R and .r for the R language??

**Comment by gnu-user** [comment]
@anvaka And if you could also add Common Lisp while you're adding Clojure I would really appreciate it! The extensions for Common Lisp are most frequently **.lisp** or **.lsp** .

**Comment by anvaka** [comment]
@rmflight I had R before. Unfortunately visualizations were not usable for R - they were made mostly from numbers. That's because I can only differentiate languages by file extension. As it turned out `.r` was also used for resource files, which looked mostly like an assembly language (thus the numbers)

**Comment by anvaka** [comment]
Added:

* [Lisp](https://anvaka.github.io/common-words/#?lang=lisp)
* [Clojure](https://anvaka.github.io/common-words/#?lang=clj)
* [Kotlin](https://anvaka.github.io/common-words/#?lang=kt)
* [CMake](https://anvaka.github.io/common-words/#?lang=cmake)
* [Swift](https://anvaka.github.io/common-words/#?lang=swift)

Please let me know if something doesn't make sense or is missing

**Comment by dooglio** [comment]
For CMake, I love that you made it in the shape of the Kitware logo!

**Comment by gnu-user** [comment]
Thanks @anvaka this looks great! While on functional languages you could also try adding Haskell, the file extension most commonly used is **.hs** or **.lhs** also in terms of rendering the word cloud Haskell has a great logo.

![](https://upload.wikimedia.org/wikipedia/commons/thumb/1/1c/Haskell-Logo.svg/200px-Haskell-Logo.svg.png)

You can find a high resolution version of the logo at the following URL: https://upload.wikimedia.org/wikipedia/commons/thumb/1/1c/Haskell-Logo.svg/2000px-Haskell-Logo.svg.png

**Comment by derekbrown** [comment]
Requesting Elixir.

**Comment by anvaka** [comment]
Added both:

* [Haskel](https://anvaka.github.io/common-words/#?lang=hs) (only .hs and .hsc extensions, as described by [linguist](https://github.com/github/linguist/blob/master/lib/linguist/languages.yml#L1736-L1747)).
* [Elixir](https://anvaka.github.io/common-words/#?lang=ex)

**Comment by ZevEisenberg** [comment]
If you add Objective-C, keep in mind that `.h` is a header format shared across multiple languages, and `.m` is shared with Matlab (I think). You might need more advanced heuristics to figure out what language a file is. There may be some good language detection in [cloc](http://cloc.sourceforge.net) that you can use for inspiration.

**Comment by ForNeVeR** [comment]
That's very nice! Could you please add support for F#? `.fs` and `.fsi` are the extensions.

**Comment by anvaka** [comment]
@ZevEisenberg yes, unfortunately I don't have anything advanced yet. I guess one could use user defined function which would classify source code text more accurately...

For now I'm just considering `.m` and `.mm` as objective c/c++ code.

@ForNeVeR the same is true about `.fs` extension - I know Forth, GLSL, and Filterscript are using it. Please let me know if you see anything suspicious...

**Comment by joneshf** [comment]
This is great! Would you be up for adding PureScript (`.purs`) and elm (`.elm`)?

**Comment by anvaka** [comment]
Added

* [Elm](https://anvaka.github.io/common-words/#?lang=elm)
* [PureScript](https://anvaka.github.io/common-words/#?lang=purs)
* [Pascal](https://anvaka.github.io/common-words/#?lang=pas)

**Comment by anvaka** [comment]
@rmflight I added R language. I used case sensitive `.R` extension and results seems better than for `.r` (which was mixed with resource files). Please let me know if something doesn't look right: 

https://anvaka.github.io/common-words/#?lang=r

**Comment by anvaka** [comment]
[Erlang is here](https://anvaka.github.io/common-words/#?lang=erl)

**Comment by rmflight** [comment]
@anvaka That actually looks like it is capturing the relevant semantics of the language! Thank you!

**Comment by bdkosher** [comment]
Awesome visualizer! I'd like to request the addition of Groovy (`.groovy`). 

The logo is quite busy, 
![Groovy logo](https://raw.githubusercontent.com/groovy/artwork/master/medium.png)
but there's a simplified version used as an icon for their console.
![Groovy Console icon](https://github.com/apache/groovy/blob/master/subprojects/groovy-console/src/main/resources/groovy/ui/ConsoleIcon.png?raw=true)

**Comment by anvaka** [comment]
@bdkosher [here it is](https://anvaka.github.io/common-words/#?lang=groovy)

**Comment by nadako** [comment]
How about Haxe (.hx)? :)
